networkx/algorithms/summarization.py

```diff
@@ -0,0 +1,187 @@
+"""
+Algorithms for reducing the number of nodes/edges in a graph
```

Comment:
If you have the energy and time, it would nice to add a bit more context here about graph summarization.  

Maybe a few sentences about graph summarization methods and applications w/ a link to:
https://dl.acm.org/doi/abs/10.1145/3186727

I would also add something about the core techniques:

    grouping methods: node and edge based
    bit compression methods: lossy and lossless
    sparsification methods
    influence based methods

And dedensification note where dedensification falls in this categorization.

Reply:
Yes, definitely. I'll do this shortly.

Reply:
Sorry for the delay.  I have made a first attempt at giving a summary of graph summarization